client/src/toc-trees.ts

```diff
@@ -0,0 +1,88 @@
+import vscode from 'vscode'
+import { LanguageClient } from 'vscode-languageclient/node'
+import { getRootPathUri, expect, constructCollectionUri, constructModuleUri } from './utils'
+import { ExtensionServerRequest, BundleTreesResponse } from '../../common/src/requests'
+import { TocTreeCollection, TocTreeElementType, TocTreeModule } from '../../common/src/toc-tree'
+
+export class TocTreesProvider implements vscode.TreeDataProvider<TocTreeItem> {
+  private readonly _onDidChangeTreeData: vscode.EventEmitter<TocTreeItem | undefined > = new vscode.EventEmitter<TocTreeItem | undefined >()
+  readonly onDidChangeTreeData: vscode.Event<TocTreeItem | undefined > = this._onDidChangeTreeData.event
+
+  constructor(private readonly client: LanguageClient) {
+  }
+
+  refresh(): void {
+    this._onDidChangeTreeData.fire(undefined)
+  }
+
+  getTreeItem(element: TocTreeItem): TocTreeItem {
+    return element
+  }
+
+  async getChildren(element?: TocTreeItem): Promise<TocTreeItem[]> {
+    if (element !== undefined) {
+      return element.children
+    }
+
+    const uri = expect(getRootPathUri(), 'No workspace root for ToC trees')
+    const bundleTrees: BundleTreesResponse = await this.client.sendRequest(
+      ExtensionServerRequest.BundleTrees,
+      { workspaceUri: uri.toString() }
```

Comment:
Typescript doesn't end up type checking this argument (since the signature of `sendRequest` allows `any`). One option would be to call a more type-safe wrapper function like [this one](https://github.com/openstax/poet/blob/73200cb45af3f687fbf59e0739d0e0aef1195a62/client/src/panel-toc-editor.ts#L59) (which could be either copied or moved to a shared location and exported)

Reply:
[Done](https://github.com/openstax/poet/pull/46/commits/f9c31dc4dd67b1821963e76991bb38cbaf5da931)!